ipatests/test_integration/test_commands.py

```diff
@@ -748,6 +748,27 @@ def test_certificate_out_write_to_file(self):
 
             self.master.run_command(['rm', '-f', filename])
 
+        # Ensure that ca/cert-show doesn't leave an empty file when
+        # the requested ca/cert does not exist.
+        commands = [
+            ['ipa', 'cert-show', '0xdeadbeef', '--certificate-out'],
+            ['ipa', 'ca-show', 'notfound', '--certificate-out'],
+        ]
+
+        for command in commands:
+            cmd = self.master.run_command(['mktemp', '--dry-run'])
+            filename = cmd.stdout_text.strip()
+
+            result = self.master.run_command(command + [filename],
+                                             raiseonerr=False)
+            assert result.returncode == 1
+
+            result = self.master.run_command(
+                ['stat', filename],
+                raiseonerr=False
+            )
+            assert result.returncode == 1
```

Comment:
According to the man page for ipa ipa xx-show should return 2 when an entry is not found: https://github.com/freeipa/freeipa/blob/adf95dcf86239f7d4145509303a01f0518134b0f/client/man/ipa.1#L199-L204
The test run shows that ipa ca-show correctly returns 2 for an unknown ca but ipa cert-show returns 1. The behavior of cert-show has been invalid maybe since the beginning (reproduced in RHEL 7.9) so I'm not sure it's worth fixing it as it would change an historical behavior.
But the test needs to be changed as ca-show correctly returns 2 if ca is not found.

Reply:
It's a two-line change to return NotFound for cert-show. The output is noticeably different:
current:
`ipa: ERROR: Certificate operation cannot be completed: Request failed with status 404: Non-2xx response from CA REST API: 404. Request ID 0x63 not found (404)`
possible:
`ipa: ERROR: Request failed with status 404: Non-2xx response from CA REST API: 404. Certificate ID 0x63 not found`

Reply:
I added a separate patch which implements the 404 test. This should fix the failing test and provide a more consistent return value.

Reply:
BTW, I'm ok with a nak on this. There could be things that rely on certain exceptions being raised beyond the cli